### DISCUSSION TITLE
Wrong sensor age count (Libre 2 plus)

### DISCUSSION THREAD

**Discussion by muccioli** [NightscoutFoundation/xDrip#3551] [Bug reports]
Good morning.
Recently in Brazil, the Libre 2 plus was launched, which my son is using. To send the data to nighscout we use juggluco. The first sensor worked fine. After placing the second sensor, the age of the sensor is being counted incorrectly, as if the other sensor was still working. The xdrip on my son’s phone is showing the date correctly, only when I access the [fly.dev](http://fly.dev/) page to monitor his blood sugar levels, the Libre’s age is wrong.
I don’t know exactly if the problem is?
![Xdrip](https://github.com/NightscoutFoundation/xDrip/assets/118318032/cc8abea1-fa97-4385-9e09-a3c6c206d5fd)

If anyone can help, I would be grateful.
Thanks.

**Comment by Navid200** [comment]
Where does xDrip show the age incorrectly?   Can you show a screenshot of xDrip instead of a different app?

**Comment by muccioli** [comment]
Em dom., 30 de jun. de 2024 10:10, Navid ***@***.***>
escreveu:

**Reply by muccioli** [reply]
![IMG_20240630_102120519](https://github.com/NightscoutFoundation/xDrip/assets/118318032/47bf7172-de97-46b3-85d1-492c9832ee98)
![Screenshot_20240630-101812](https://github.com/NightscoutFoundation/xDrip/assets/118318032/0b034ec2-487b-40fb-8901-98832fab2b51)

**Reply by Navid200** [reply]
Please don't reply to an email to post here.
  
I have no idea what you intend to say in your email reply.  
Please log into GitHub and go to the discussion so that you can see what your post looks like.

**Reply by Navid200** [reply]
So, xDrip shows the correct age.  But, Nightscut shows an incorrect age, right?
What uploads to Nightscout?

**Reply by muccioli** [reply]
To send the data to nighscout we use juggluco

**Reply by Navid200** [reply]
So, do I understand you correctly that it is Juggluco that needs to be fixed and not xDrip?

**Reply by muccioli** [reply]
Juggluco is correct!
![IMG_20240630_174754506](https://github.com/NightscoutFoundation/xDrip/assets/118318032/b853d12c-b41f-46d7-9968-82841823366a)

**Reply by Navid200** [reply]
As far as I know if xDrip uses the companion app mode or any other way for getting data from another app, it may not be possible for xDrip to know the correct sensor expiry time unless the other app provides it.  I don't believe Juggluco broadcasts anything to xDrip.

**Reply by muccioli** [reply]
My friend, whether you believe it or not doesn't make much difference, stick to the facts and the photos that prove it. With the first sensor everything was working normally. As a developer, you should seek a solution and whether or not this is possible. There is no basis in...I don't believe it.

**Reply by Navid200** [reply]
I label this as Libre and a developer with experience with Libre can help.

**Reply by muccioli** [reply]
Librelink is correct too!

**Reply by Navid200** [reply]
@PhilTU

**Reply by PhilTU** [reply]
I think Juggluco's extended end time has the wrong data, should surely be Sun 14/7/24 02:00

xDrip+ has the correct start time, matching juggluco. Can you show us the Libre 2 status page from xDrip+ (swipe sideways from classic) or somewhere else in xDrip+ that shows expiry time or sensor age.

What is the IddS field in the Nightscout view ? Couldn't see it in the docs.

If juggluco is uploading to Nightscout then xDrip isn't responsible for the sensor age / life expectancy supplied to Nightscout, juggluco is.

The Libre patched input facility on xDrip+ is being sent data by juggluco acting as if it were the patched Librelink app. The Patched Librelink app predates the Libre2+ sensor by some years and may not be capable of handling different sensor lives other than 14 days.
